prody/proteins/interactions.py

```diff
@@ -885,29 +900,53 @@ def calcHydrogenBondsDCD(atoms, trajectory, distA=3.0, angle=40, cutoff_dist=20,
         except TypeError:
             raise TypeError('coords must be an object '
                             'with `getCoords` method')
-        
-    if isinstance(trajectory, Atomic):
-        trajectory = Ensemble(trajectory)
-                        
+
     HBs_all = []
+
+    if trajectory is not None: 
+        if isinstance(trajectory, Atomic):
+            trajectory = Ensemble(trajectory)
+            
+        trajectory.reset()
         
-    for j0, frame0 in enumerate(trajectory):  
-        LOGGER.info('Frame: {0}'.format(j0))
-        protein = atoms.select('protein')
-        hydrogen_bonds = calcHydrogenBonds(protein, distA, angle, cutoff_dist, **kwargs)
-        HBs_all.append(hydrogen_bonds)
-        
+        for j0, frame0 in enumerate(trajectory):  
+            LOGGER.info('Frame: {0}'.format(j0))
+            protein = atoms.select('protein')
+            hydrogen_bonds = calcHydrogenBonds(protein, distA, angle, cutoff_dist, **kwargs)
+            HBs_all.append(hydrogen_bonds)
+    
+    else:
+        if atoms.numCoordsets() > 1:
+            for i in range(len(atoms.getCoordsets())):
+                LOGGER.info('Model: {0}'.format(i))
+                atoms.setACSIndex(i) 
+                protein = atoms.select('protein')
+                hydrogen_bonds = calcHydrogenBonds(protein, distA, angle, cutoff_dist, **kwargs)
+                HBs_all.append(hydrogen_bonds)
+        else:
+            LOGGER.info('Include trajectory or use multiple PDB file.')
+            
     return HBs_all
 
 
-def calcSaltBridgesDCD(atoms, trajectory, distA=4.5):
+def calcSaltBridgesDCD(atoms, trajectory=None, distA=4.5, **kwargs):
```

Comment:
again, not dcd but something else and explain. I'd still suggest supporting an Ensemble too

Reply:
probably you do support Ensemble but you need to have docs and a type check

Reply:
Yes, I do support Ensemble. If you use PDB with multiple frames as atoms and do not include trajectory. It will compute the results for multiple frames.

Reply:
No, I meant actually providing an Ensemble as input. I haven't tested it but I'm pretty sure it should work if you can convert the atoms to one inside

Reply:
You could have something like the following
```
        if isinstance(trajectory, Atomic):
            trajectory = Ensemble(trajectory)
        elif not isinstance(Trajectory, (Trajectory, Ensemble):
            raise TypeError('Trajectory should be an Atomic, Trajectory or Ensemble object, or None')
```